packages/theme/src/unified/UnifiedThemeProvider.tsx

```diff
@@ -35,6 +39,14 @@ export interface UnifiedThemeProviderProps {
   noCssBaseline?: boolean;
 }
 
+// Background at https://mui.com/x/migration/migration-data-grid-v4/#using-mui-core-v4-with-v5
+// Rather than disabling globals and custom seed, we instead only set a production prefix that
+// won't collide with MUI 5 styles. We've already got a separate class name generator for v5 set
+// up in MuiClassNameSetup.ts, so only the production JSS needs deduplication.
+const generateV4ClassName = createGenerateClassName({
+  productionPrefix: 'jss4-',
+});
```

Comment:
I switched over to just the `productionPrefix` approach rather than what's recommended in docs. It seems like this has the same effect in combination with the v5 class generator override, with much less impact. Not sure if there's still a risk of collisions with this approach but always keen to avoid breaking existing code. My main worry with this approach is that there would be conflicts for classes like `Mui-disabled`, but I would also expect those to never apply styles alone, potentially removing any conflicts.

CC @oliviertassinari @mnajdova in case you have any insights on this approach and whether it's worth adding to docs if it's a proper alternative